thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/DimensionAnalysisPipeline.java

```diff
@@ -0,0 +1,195 @@
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Pipeline for identifying relevant dimensions by performing
+ * contribution analysis. The pipeline first fetches the Current and Baseline entities and
+ * MetricEntities in the search context. It then maps the metrics to ThirdEye's internal database
+ * and performs contribution analysis using a {@code DimensionScorer).
+ *
+ * @see DimensionScorer
+ */
+public class DimensionAnalysisPipeline extends Pipeline {
+  private static final Logger LOG = LoggerFactory.getLogger(DimensionAnalysisPipeline.class);
+
+  public static final String PROP_PARALLELISM = "parallelism";
+
+  public static final long TIMEOUT = 120000;
+
+  private static final String KEY = "key";
+  private static final String DIMENSION = DimensionScorer.DIMENSION;
+  private static final String VALUE = DimensionScorer.VALUE;
+  private static final String COST = DimensionScorer.COST;
+
+  private final MetricConfigManager metricDAO;
+  private final DatasetConfigManager datasetDAO;
+  private final DimensionScorer scorer;
+  private final ExecutorService executor;
+
+  /**
+   * Constructor for dependency injection
+   *
+   * @param name pipeline name
+   * @param inputs pipeline inputs
+   * @param metricDAO metric config DAO
+   * @param datasetDAO dataset config DAO
+   * @param scorer dimension scorer for contribution analysis
+   * @param executor executor service for parallel task execution
+   */
+  public DimensionAnalysisPipeline(String name, Set<String> inputs, MetricConfigManager metricDAO,
+      DatasetConfigManager datasetDAO, DimensionScorer scorer, ExecutorService executor) {
+    super(name, inputs);
+    this.metricDAO = metricDAO;
+    this.datasetDAO = datasetDAO;
+    this.scorer = scorer;
+    this.executor = executor;
+  }
+
+  /**
+   * Alternate constructor for use by PipelineLoader
+   *
+   * @param name pipeline name
+   * @param inputs pipeline inputs
+   * @param properties configuration properties ({@code PROP_PARALLELISM})
+   */
+  public DimensionAnalysisPipeline(String name, Set<String> inputs, Map<String, String> properties) {
+    super(name, inputs);
+
+    this.metricDAO = DAORegistry.getInstance().getMetricConfigDAO();
+    this.datasetDAO = DAORegistry.getInstance().getDatasetConfigDAO();
+    this.scorer = new DimensionScorer(ThirdEyeCacheRegistry.getInstance().getQueryCache());
+
+    int parallelism = Integer.parseInt(properties.get(PROP_PARALLELISM));
+    this.executor = Executors.newFixedThreadPool(parallelism);
+  }
+
+  @Override
+  public PipelineResult run(PipelineContext context) {
+    Set<MetricEntity> metricsEntities = context.filter(MetricEntity.class);
+
+    final TimeRangeEntity current = TimeRangeEntity.getContextCurrent(context);
```

Comment:
I think default values should be the responsibility of the logic using the pipeline. Otherwise, an omission by the programmer can lead to very confusing results for the user
For example, a UI may display a Wo4W comparison but not pass on this baseline. The RCA framework would execute without warning and provide misleading WoW results.